### DISCUSSION TITLE
Lesson 12 ether and ether in  testGetTokenAmountFromUsd()

### DISCUSSION THREAD

**Discussion by bigBagBoogy** [Cyfrin/foundry-full-course-cu#414] [General]
I Actually broke my brain on this one, so I'd like to share this:

in the function below there's actually twice the word "ether", but with different meanings.

chatGPT3 about this:

You are absolutely correct. In the test provided, the word "ether" is used in two different contexts, and it can be confusing. Let's clarify the meanings of each occurrence:

uint256 expectedWeth = 0.05 ether;: In this line, the word "ether" refers to the native cryptocurrency of the Ethereum blockchain, commonly denoted as ETH. It represents 0.05 ETH, which is 0.05 times the base unit of Ethereum (Wei). This value is expressed in terms of the native cryptocurrency.

uint256 amountWeth = dsce.getTokenAmountFromUsd(weth, 100 ether);: In this line, the word "ether" is used as a unit to represent the equivalent value of $100 in the stablecoin (DSC) called "wholeCoin." It is used to pass the value denominated in the stablecoin (DSC) to the function getTokenAmountFromUsd. The function expects the value of USD in Wei equivalent to $100, expressed as 100 * 1e18, where 1e18 represents 1 wholeCoin (the smallest unit of DSC). So, in this context, "ether" does not refer to the native cryptocurrency (ETH); it represents the stablecoin (DSC) unit "wholeCoin."

To avoid confusion, it would be better to use more descriptive variable names and comments in the test to indicate the units being used.

Also my VScode picks up ```ether``` as a global variable of the EVM or ETH. (I see it does not so here in Github.

Have a perfect day,

bigBagBoogy

```solidity
function testGetTokenAmountFromUsd() public {
    // If we want $100 of WETH @ $2000/WETH, that would be 0.05 WETH
    uint256 expectedWeth = 0.05 ether; // this ether here is Ethereum ether
    uint256 amountWeth = dsce.getTokenAmountFromUsd(weth, 100 ether); // 100 ether here is $100 worth of the 1 - 1 stablecoins (called "ether" here) and this is not 100 ETH!!!!
    assertEq(amountWeth, expectedWeth);
}```

**Comment by shikhar229169** [comment]
Boom!

Yeah! ether just multiplies the base number by 1e18.
